File: ga/4omini_result_2024_7/result/macro_func_mini/14_macro_func.cpp
use std::io;

fn cpp_max(x: i32, y: i32) -> i32 {
    if (x + y) < 5 {
        y
    } else {
        x
    }
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");

    let numbers: Vec<i32> = input
        .trim()
        .split_whitespace()
        .map(|s| s.parse().expect("Please enter a valid number"))
        .collect();

    let (a, b) = (numbers[0], numbers[1]);

    let result = cpp_max(a, b);
    println!("Result: {}", result);
}

File: ga/glm4_result_2024_6/result/macro_func_glm/14_macro_func.rs
fn main() {
    let mut input = String::new();
    println!("Enter two numbers separated by space: ");
    std::io::stdin().read_line(&mut input).expect("Failed to read line");
    let mut numbers = input.trim().split_whitespace();
    let a: i32 = numbers.next().unwrap().parse().expect("Invalid number");
    let b: i32 = numbers.next().unwrap().parse().expect("Invalid number");

    let result = if (a + b) < 5 { b } else { a };
    println!("Result: {}", result);
}